x/superfluid/keeper/stake.go

```diff
@@ -464,10 +473,10 @@ func (k Keeper) alreadySuperfluidStaking(ctx sdk.Context, lockID uint64) bool {
 }
 
 // mintOsmoTokensAndDelegate mints osmoAmount of OSMO tokens, and immediately delegate them to validator on behalf of intermediary account.
-func (k Keeper) mintOsmoTokensAndDelegate(ctx sdk.Context, osmoAmount osmomath.Int, intermediaryAccount types.SuperfluidIntermediaryAccount) error {
+func (k Keeper) mintOsmoTokensAndDelegate(ctx sdk.Context, osmoAmount osmomath.Int, intermediaryAccount types.SuperfluidIntermediaryAccount) (newShares math.LegacyDec, err error) {
```

Comment:
We have always not have had any event emissions upon refresh and increase right? In fact, Im not sure if we have incidences where non msg actions has emitted events.

Reply:
We can start emitting events in Increase as you have mentioned, but I would address that in a separate PR, we would have to revisit all the existing methods to  determine if it should emit event or not + I think the inconsistency where some methods emits events whilst some doesn't would cause more confusion vs only emitting it upon initial delegation

Reply:
Yeah sounds good, thanks